logos/logos-orchestrator/src/logos/request_content.py

```
@@ -0,0 +1,297 @@
+def _install_request_size_limit(request: Request) -> None:
+    """Reject declared and chunked multipart bodies before unbounded spooling."""
+    content_length = request.headers.get("content-length")
+    if content_length:
+        try:
+            if int(content_length) > MAX_AUDIO_REQUEST_BYTES:
+                raise HTTPException(
+                    status_code=413,
+                    detail=f"Audio request exceeds the {MAX_AUDIO_REQUEST_BYTES}-byte request limit",
+                )
+        except ValueError:
+            raise HTTPException(status_code=400, detail="Invalid Content-Length header")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)